### DISCUSSION TITLE
middleware for server-side authentication

### DISCUSSION THREAD

**Discussion by MuttakinHasib** [vercel/next.js#39447] [Help]
I want to implement server-side authentication based on cookies. I implemented it this way. Is it ok?

**1. How can I check whether the cookie is expired or not?**
**2. How can I implement it in stable middleware?**
> pages/dasboard/_middleware.tsx

```jsx
import { NextRequest, NextResponse } from 'next/server'
import { SESSION_ID } from '../constants'

export function authMiddleware(req: NextRequest) {
  const isLoggedIn = req.cookies[SESSION_ID] || false
  if (isLoggedIn) {
    return NextResponse.next()
  }
  // Update url pathname
  req.nextUrl.pathname = `/${isLoggedIn ? 'dashboard' : 'login'}`

  // Rewrite to url
  return NextResponse.redirect(req.nextUrl)
}

```

**2. how can I apply this middleware only for dashboard routes?**

📦src
 ┣ 📂pages
 ┃ ┣ 📂dashboard
 ┃ ┃ ┣ 📂account
 ┃ ┃ ┃ ┗ 📜index.tsx
 ┃ ┃ ┣ 📂support
 ┃ ┃ ┃ ┗ 📜index.tsx
 ┃ ┃ ┗ 📜 _middleware.tsx
 ┃ ┃ ┗ 📜index.tsx
 ┃ ┣ 📜_app.tsx
 ┃ ┗ 📜index.tsx

@leerob Please help me

**Comment by MuttakinHasib** [comment]
@icyJoseph can you please help me ?

**Reply by goacgras** [reply]
Have you solved this?

**Comment by abusada** [comment]
@MuttakinHasib I think you are looking for something like [this example](https://github.com/vercel/examples/blob/main/edge-functions/jwt-authentication/middleware.ts.) 

It seems that you are using the old middleware, check [here](https://nextjs.org/docs/messages/middleware-upgrade-guide) for stable middleware upgrade guide

**Reply by MuttakinHasib** [reply]
@abusada  How can I call an API in next.js middleware by sending a cookie?

**Comment by MuttakinHasib** [comment]
@icyJoseph @abusada  @leerob 
### **When I'm trying to build, I'm getting this error.**

![image](https://user-images.githubusercontent.com/44552983/183907589-a73b6829-944e-481c-b14a-af3d4fac1869.png)

**Comment by DonHuskini** [comment]
@MuttakinHasib here's an example. Use the matcher key. https://github.com/gabrielmlinassi/gym-lp/blob/main/src/middleware.ts
